### PR TITLE
Add SM-T590 (gta2xlwifi) support

### PR DESCRIPTION
**Pull request by quent1-fr**
Add support for the Samsung Galaxy Tab A2 XL WiFI (2018) Rev05.

### CODE REVIEW

File: dts/sdm450-samsung-r05.dts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	
+	gta2xlwifi-generic {
+		model = "Samsung Tab A2 XL WIFI Rev05";
+		compatible = "samsung,gta2xlwifi-eur", "qcom,sdm450", "qcom,msm8953", "lk2nd,device";
+		lk2nd,match-bootloader = "T590*";
+		lk2nd,pstore = <0x86100000 0x100000>;
+	};
+	
+	// Bootloader won't continue if it can't delete some nodes from below
+	soc {
```
Comment (alikates) [inline dts/sdm450-samsung-r05.dts:20]: Hi, can you test if it still works without this? It should only be necessary on some motorola phones.

Comment (vldly) [reply dts/sdm450-samsung-r05.dts:20]: @alikates Actually it's other way around, everything except Samsung SDM450 boards probably doesn't need this.

File: dts/sdm450-samsung-r05.dts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/include/ "msm8953.dtsi"
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <351 0x0>;
+	qcom,board-id = <8 5>;
+	
+	gta2xlwifi-generic {
+		model = "Samsung Tab A2 XL WIFI Rev05";
+		compatible = "samsung,gta2xlwifi-eur", "qcom,sdm450", "qcom,msm8953", "lk2nd,device";
```
Comment (lucasredelive) [inline dts/sdm450-samsung-r05.dts:14]: remove "qcom,sdm450"
its broke another devices with this same soc

File: dts/sdm450-samsung-r05.dts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		ranges = <0x0 0x0 0x0 0xffffffff>;
+
+		qcom,memshare {
+			qcom,client_4 {
+				qcom,peripheral-size = <0x6000000>;
+			};
+		};
+	};
+
+	reserved-memory {
```
Comment (lucasredelive) [inline dts/sdm450-samsung-r05.dts:32]: need to delete some nodes, or the kernel will not be able to start because the memory is already being allocated

**Review by lucasredelive — CHANGES_REQUESTED** [review]

**Pull request merged**